### DISCUSSION TITLE
How to use custom-type to replace base-type such as `number_integer_t`?

### DISCUSSION THREAD

**Discussion by Kael-Chan** [nlohmann/json#3789] [Q&A]
For some reason, default type int64_t/uint64_t for NumberIntegerType/NumberUnsignedType is not big enough to use. And replacing by string can not be admitted.

Can I replace them with some user-defined big number class?

- If yes, what interfaces should I support in those custom class, to make my editing transforming and the JSON module would run as usual?

- If no, is there any charming way to use nlohmann/json with big number? 

thank,
Kael.

**Comment by nlohmann** [comment]
Related: https://github.com/nlohmann/json/discussions/2228

**Reply by Kael-Chan** [reply]
Thanks, but a specific `int***_t` type still has a pre-defined limiting, no matter how big the "star"s are.

What I need for `NumberIntegerType/NumberUnsignedType` is an unlimited big number class (may called `BigNumber`). when loading from json, both of `number-1234` and `number-factorial-of-1234` (more than 3k digits) can be parsed to an object of this class.

*Maybe another solution:*

Simply using `to_json/from_json` func pair will make json number string with quotation marks. So another way to achieve this goal is customing the serialization of `BigNumber` **without quotation marks**.

**Reply by falbrechtskirchinger** [reply]
There's #3577, but it appears to be abandoned and I had some issues with it.
